GEOWAVE-306

rwgdrummer opened the pull request:
#306

chrisbennight commented on geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/clustering/CentroidManagerGeoWave.java:
```diff
@@ -283,12 +291,23 @@ public CentroidManagerGeoWave(
+					runTimeProperties.getProperty(GlobalParameters.Global.ACCUMULO_PASSWORD),
+					runTimeProperties.getProperty(GlobalParameters.Global.ACCUMULO_NAMESPACE));
+		}
+		catch (InstantiationException | IllegalAccessException e) {
```
What's the intent behind converting these exceptions all to IOException?

rwgdrummer replied:
Used in situations such as setup in Mapper/Reducers that only handles IOException.  Rather littering the method signature and pushing the redundant handling else where, I do it here.

chrisbennight replied:
Gotcha - I understand it in the case of a constrained method that needs to be implemented/overrided - just didn't see that here.   What does e.getMessage() do in this case?   Is there any value in something like  
```Java
throw new IOException(e, e.getMessage())
```
(might be pointless - intent here is just to make sure we aren't slightly obfuscating the InstatiantionException of IllegalAcessException, since those are more useful than IOException)

rwgdrummer replied:
yep.

chrisbennight commented on geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/PropertyManagement.java:
```diff
@@ -68,46 +77,40 @@ public Object get(
-
-			return properties.get(pName);
+		if (!properties.containsKey(pName)) {
+			LOGGER.info("Setting parameter : " + pName + " to " + value.toString());
```
Instead of string concatenation, move the LOGGER statements over to use dynamic content injection {}
ref: http://slf4j.org/faq.html#logging_performance
String won't even be built if the log level is high enough.

jwomeara replied:
Nice!  Didn't even know you could do that.
On Apr 2, 2015 10:16 PM, "Chris Bennight" <notifications@github.com> wrote:

> In
> geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/PropertyManagement.java
> <https://github.com/ngageoint/geowave/pull/307#discussion_r27712447>:
>
> >  			final ParameterEnum propertyName,
> >  			final Object value ) {
> >  		final String pName = toPropertyName(propertyName);
> > -		synchronized (MUTEX) {
> > -			if (!properties.containsKey(pName)) {
> > -				LOGGER.info("Setting parameter : " + pName + " to " + value.toString());
> > -				properties.put(
> > -						pName,
> > -						value);
> > -				return value;
> > -			}
> > -
> > -			return properties.get(pName);
> > +		if (!properties.containsKey(pName)) {
> > +			LOGGER.info("Setting parameter : " + pName + " to " + value.toString());
>
> Instead of string concatenation, move the LOGGER statements over to use
> dynamic content injection {}
> ref: http://slf4j.org/faq.html#logging_performance
> String won't even be built if the log level is high enough.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ngageoint/geowave/pull/307/files#r27712447>.
>

rwgdrummer replied:
True,  Can do this.  I know SLF4j supports this.

chrisbennight replied:
And to be fair we still have to move over the rest of the codebase to this as well.

rfecher replied:
it sounds like someone is going to be padding his github stats this weekend ;)

chrisbennight commented on geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/PropertyManagement.java:
```diff
@@ -68,46 +77,40 @@ public Object get(
-					toPropertyName(propertyName),
-					value);
-		}
+		properties.put(
```
Minor nit, but probably should be properties.setProperty()
(I think all it does is enforce (String, String), but a good habbit)

rwgdrummer replied:
Then I will change it to a map.  The values need to serializable objects.  I will make the adjustment.

chrisbennight replied:
That works to!

rwgdrummer replied:
Thanks.

chrisbennight commented on geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/PropertyManagement.java:
```diff
@@ -399,21 +419,19 @@ public DistributableQuery getPropertyAsQuery(
+			if (!option.hasArg()) {
+				properties.put(
+						option.getLongOpt(),
+						true);
```
Not sure what will happen here - not supposto have anything but string, string in a properties, but this snuck in using the put method.  Might implicitly get converted, but probably better to be explicit.

rwgdrummer commented:
I started looking at this again more closely.
PropertyManagement changed a bit for me as I started seeing its potential as a uniform way to handle parameters in the analytics. Looking at it some more tonight, I think there is some thought that I left unfinished.  I am going to make some additional adjustments tomorrow.

rwgdrummer commented:
Addressed the concerns of the last push.
(1) Added localized message to IOException
(2) Consistent handling of serializable objects in the parameter management.
(3) formatting efficiencies.
So, ready for another review.
Thanks
